torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -2186,6 +2206,11 @@ def _run_test_profiler_with_async_rpc_udf(self):
         self._profiler_test_with_rpc(RPCExecMode.ASYNC, my_sleep_func, args=(1,))
         self._profiler_test_with_rpc(RPCExecMode.ASYNC, my_sleep_func, args=(1,),
                                      use_record_function=True)
+        # Test to ensure that kineto profiler enabled in RPC does not enable
+        # RPC profiling (it is unsupported) and does not result in issues.
+        self._profiler_test_with_rpc(
+            RPCExecMode.ASYNC, my_sleep_func, args=(1,), kineto_profile=True
+        )
 
     @dist_init
     def test_profiler_with_async_rpc_udf(self):
```

Comment:
Unrelated, curious about this test. Why does this and the test below it call the same func?

Reply:
Hmm the single threaded one was supposed to test RPC with num_worker_threads=1. Perhaps that went away in the tensorpipe refactor or somehow?